cncf-youtube-channel-summarizer/transcript_summarizer.py

```diff
@@ -0,0 +1,120 @@
+        texts = text_splitter.create_documents([transcript])
+        map_summary_template = open('prompt/chunks_summary_prompt1.txt').readlines()
+        map_summary_template = ''.join(map_summary_template)
+        combine_summary_template = open('prompt/combine_summary_prompt.txt').readlines()
+        combine_summary_template = ''.join(combine_summary_template)
+
+        keyword_template = open('prompt/keyword_template.txt').readlines()
+        keyword_template = ''.join(keyword_template)
+
+        map_prompt = PromptTemplate(template=map_summary_template, input_variables=["text"])
+        combine_prompt = PromptTemplate(template=combine_summary_template, input_variables=["text"])
+        prompt_keywords = PromptTemplate(template=keyword_template, input_variables=["text"])
+        chain_summary = load_summarize_chain(llm=llm_summary, chain_type="map_reduce", map_prompt=map_prompt,
+                                                 combine_prompt=combine_prompt, verbose=True)
+        summary = chain_summary.run(texts)
+        summary_doc = text_splitter.create_documents([summary])
+        chain_keywords = load_summarize_chain(llm=llm_keywords, chain_type="stuff", prompt=prompt_keywords,
+                                              verbose=False)
+        keywords = chain_keywords.run(summary_doc)
+        return summary, keywords
+
+    @retry(stop=stop_after_attempt(3), wait=wait_fixed(2))
+    def run(self):
+        i = 0
+        missed_video_id = []
+        print(self.llm_summary.get_details())
+
+        max_sequence_length = self.llm_summary.get_details()['model_limits']['max_sequence_length']
+        chunk_size = max_sequence_length - 1000
+        chunk_overlap = 50
+        cncf_video_summary = pd.DataFrame(columns=['video_id', 'video_title', 'summary', 'keywords'])
+        cncf_video_summary.to_csv('cncf_video_summary.csv', index=False)
+
+        for key in self.videos_dict.keys():
+            transcript = self.videos_dict[key]['transcript'][:100]
+            print(len(transcript.split(' ')))
+            try:
+                summary, keywords = self.LLM_summarizer(self.llm_summary.to_langchain(), self.llm_keywords.to_langchain(), transcript,
+                                               chunk_size, chunk_overlap, key)
+            except:
+                logger.error(f"Failed to generate the summary and keywords for video: {key}")
+                missed_video_id = open('missed_video_id.txt', 'a')
+                missed_video_id.write(key)
+                continue
+
+            data = {'video_id': [key], 'video_title': [self.videos_dict[key]['video_title']],
+                    'conference_name': [self.videos_dict[key]['play_list']['title']], 'summary': [summary],
+                    'keywords': [keywords]}
+            df = pd.DataFrame(data)
+            df.to_csv('cncf_video_summary.csv', mode='a', index=False, header=False)
```

Comment:
it will be great if you can also post the initial summary here so we can see what the contents look like.